Runtime/Highlight/ObjectHighlighter.cs

```diff
@@ -0,0 +1,457 @@
+﻿using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Innoactive.Creator.Unity;
+
+namespace Innoactive.Creator.BasicInteraction
+{
+    /// <summary>
+    /// Adds highlighting functionality to a GameObject with Renderers.
+    /// </summary>
+    /// <remarks>
+    /// Highlights are always queue following a LIFO (Last In First Out) scheme. 
+    /// </remarks>
+    public class ObjectHighlighter : MonoBehaviour
+    {
+        private readonly Dictionary<string, Material> highlightIDs = new Dictionary<string, Material>();
+
+        protected bool IsBeingHighlighted => highlightIDs.Any();
+
+        [SerializeField, HideInInspector]
+        private Renderer[] renderers = {};
+        
+        [SerializeField, HideInInspector]
+        private MeshRenderer highlightMeshRenderer = null;
+        
+        [SerializeField , HideInInspector]
+        private MeshFilter highlightMeshFilter = null;
+        
+        protected virtual void Reset()
+        {
+            RefreshCachedRenderers();
+        }
+
+        protected virtual void OnEnable()
+        { }
+
+        protected virtual void OnDisable()
+        {
+            if (IsBeingHighlighted)
+            {
+                ReenableRenderers();
+                highlightIDs.Clear();
+            }
+        }
+        
+        
+        /// <summary>
+        /// Returns true if there is any highlight active.
+        /// </summary>
+        /// <param name="listOfActiveHighlights">In case there is any active highlight, this value will be filled with all available highlights IDs.</param>
```

Comment:
```suggestion
        /// <param name="listOfActiveHighlights">In case there is any active highlight, this value will be filled with all available highlight IDs.</param>
```